### DISCUSSION TITLE
crt file created by strongswan-pki: Windows: This certificate has an invalid digital signature

### DISCUSSION THREAD

**Discussion by weiqi-chen** [strongswan/strongswan#2366] [Q&A] [answered]
I'm playing with strongSwan. I want to setup a vpn server( IKEv2), this is my ultimate goal.
I'm new to strongswan and openssl, but I can search and ask AI to slove problems.

I setup rpi-ubuntu 22.04 and vm-ubuntu 22.04（virtual machine)， they in the same lan(192.168.1.0/24).

```
root@vm-ubuntu:~/cert# apt list --installed | grep -E 'strongswan|charon'
charon-systemd/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed]
libcharon-extauth-plugins/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed,automatic]
libstrongswan-extra-plugins/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed]
libstrongswan-standard-plugins/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed]
libstrongswan/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed,automatic]
strongswan-charon/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed,automatic]
strongswan-libcharon/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed,automatic]
strongswan-pki/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed]
strongswan-starter/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed]
strongswan-swanctl/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 amd64 [installed]
strongswan/jammy-updates,jammy-security,now 5.9.5-2ubuntu2.3 all [installed]

systemctl mask strongswan-starter.service
systemctl enable strongswan.service
systemctl start strongswan.service
```
Documents I tried step by step.
- https://docs.strongswan.org/docs/5.9/pki/pkiQuickstart.html
- https://docs.strongswan.org/docs/5.9/config/quickstart.html

I managed  to "Roadwarrior Case with EAP-MSCHAPv2" so far so good.

Now I want to add another machine vm-windows 7, Now things would get complicated.
Acorrding to document, I created the ca by commands:
```
pki --gen --type ed25519 --outform pem > strongswanKey.pem

pki --self --ca --lifetime 3652 --in strongswanKey.pem \
           --dn "C=CH, O=strongSwan, CN=strongSwan Root CA" \
           --outform pem > strongswanCert.pem

pki --print --in strongswanCert.pem
```

`scp vm-ubuntu.lan:/path/strongswanCert.pem vm-windows7.lan:/path/to/desktop/strongswanCert.crt`

I open this file, I see (same picture on windows 10 and windows 7) :

![1](https://github.com/user-attachments/assets/5b97d9e4-20ba-4cba-a4ae-8910c7569cc7)
![2](https://github.com/user-attachments/assets/191334a8-8131-4d8d-a2cb-b51e9ae9641d)

I might need to recreated ca and all certs by using openssl, buy openssl is so complicated than strongswan-pki.

So is something I done wrong?

**Comment by tobiasbrunner** [comment, marked as answer]
> pki --gen --type ed25519 --outform pem > strongswanKey.pem

This creates an EdDSA key, which is a relatively modern algorithm that Windows (and many other systems) currently doesn't support. You might want to try a different key type like RSA or ECDSA (RSA is the most compatible).

**Reply by weiqi-chen** [reply]
`pki --gen --type ecdsa --size 256 --outform pem` works!
